### PR TITLE
Front-end HW-4

### PR DESCRIPTION
**Pull request by javdad**
backend:
removed expenses and incomes and add transactions. Changed and fixed some parts of code.

frontend:
added accounts, categories, transactions components and linked to the server. Statistics and admin page will be added.
fixed routes, added some styles. Working on styles. Most of 'any' types will be changed.

### CODE REVIEW

File: Client/src/app/app-routing.module.ts [hunk 1]
```diff
@@ -3,16 +3,26 @@ import { RouterModule, Routes } from '@angular/router';
...
 import { MainLayoutComponent } from './layouts/main-layout/main-layout.component';
+import { SidenavComponent } from './layouts/main-layout/sidenav/sidenav.component';
 import { NotFoundComponent } from './not-found/not-found/not-found.component';
 
 const routes: Routes = [
   {path: '', component: AuthLayoutComponent, children: [
       {path: '', redirectTo: 'login', pathMatch: 'full'},
       {path: 'login', component: AuthLoginComponent}
+    ]},
+    {path: 'main', component: MainLayoutComponent, canActivate: [AuthGuard], children: [
```
Comment (AudriusBaltusis) [inline Client/src/app/app-routing.module.ts]: Code formatting is awkward. Please use
```
{
  path: 'main',
  component: MainLayoutComponent,
  canActivate: [AuthGuard],
  children: [
    {
      path: '',
      component: SidenavComponent,
      canActivate: [AuthGuard],
      children: [
        { path: '', component: TransactionsComponent },
        { path: 'categories', component: CategoriesComponent },
        { path: 'statistics', component: StatisticsComponent },
      ]
    }
  ]
}
```

File: Client/src/app/components/account/account.component.ts [hunk 2]
```diff
@@ -0,0 +1,12 @@
+import { Component, Input } from '@angular/core';
+
+@Component({
+  selector: 'app-account',
+  templateUrl: './account.component.html',
+  styleUrls: ['./account.component.scss']
+})
+export class AccountComponent  {
+
+  @Input() account: any;
```
Comment (AudriusBaltusis) [inline Client/src/app/components/account/account.component.ts]: `any`

File: Client/src/app/components/categories/categories.component.ts [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+import { CategoryService } from 'src/app/services/category.service';
+
+@Component({
+  selector: 'app-categories',
+  templateUrl: './categories.component.html',
+  styleUrls: ['./categories.component.scss']
+})
+export class CategoriesComponent implements OnInit {
+
+  categories: any = []
```
Comment (AudriusBaltusis) [inline Client/src/app/components/categories/categories.component.ts]: `any`

File: Client/src/app/components/categories/categories.component.ts [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+
+  constructor(private category: CategoryService) { }
+
+  ngOnInit(): void {
+    this.category.getCategories().subscribe((categories) => {
```
Comment (AudriusBaltusis) [inline Client/src/app/components/categories/categories.component.ts:26]: You have to unsubscribe, or use `takeUntil()`

File: Client/src/app/components/statistics/statistics.component.ts [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+@Component({
+  selector: 'app-statistics',
+  templateUrl: './statistics.component.html',
+  styleUrls: ['./statistics.component.scss']
+})
+export class StatisticsComponent implements OnInit {
+
+  constructor() { }
+
+  ngOnInit(): void {
```
Comment (AudriusBaltusis) [inline Client/src/app/components/statistics/statistics.component.ts]: Please remove redundant `ngOnInit`

File: Client/src/app/components/transactions/transaction/transaction.component.ts [hunk 6]
```diff
@@ -0,0 +1,12 @@
+import { Component, Input } from '@angular/core';
+
+@Component({
+  selector: 'app-transaction',
+  templateUrl: './transaction.component.html',
+  styleUrls: ['./transaction.component.scss']
+})
+export class TransactionComponent  {
+
+  @Input() transaction: any;
```
Comment (AudriusBaltusis) [inline Client/src/app/components/transactions/transaction/transaction.component.ts:10]: `any`

File: Client/src/app/services/account.service.ts [hunk 7]
```diff
@@ -0,0 +1,15 @@
...
+import { Injectable } from '@angular/core';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class AccountService {
+
+  constructor(private http: HttpClient){}
+
+  getAccounts() {
```
Comment (AudriusBaltusis) [inline Client/src/app/services/account.service.ts]: Please add return type

**Review by AudriusBaltusis — CHANGES_REQUESTED** [review]

**Comment by javdad** [comment]
I made the requested changes last week. Wanted to ask if you checked them?

**Comment by javdad** [comment]
I made the requested changes last week. Wanted to ask if you checked them?

**Review by AudriusBaltusis — APPROVED** [review]

**Pull request merged**